packages/patterns/src/constants.js

```diff
@@ -9,7 +9,6 @@ export const PATTERN_CORE_SOURCES = [
 	'core',
 	'pattern-directory/core',
 	'pattern-directory/featured',
-	'pattern-directory/theme',
```

Comment:
Removing this makes the `PATTERN_CORE_SOURCES` variable name no longer true, so perhaps the variable needs to be renamed as well (`EXCLUDED_LIBRARY_SOURCES`?), or the filtering could be done a different way.

Reply:
I think it actually applies better now as previously it included theme specified patterns from the directory which are essentially theme patterns rather than core, not it is more strictly just core patterns that are not related to the current theme at all.

Reply:
I agree with @talldan on the naming here.

> previously it included theme specified patterns from the directory which are essentially theme patterns rather than core

The source of the patterns is core and the pattern directory, not the theme. The theme doesn't create or define the patterns, it only "opts into" using them, which doesn't change their actual source.

Reply:
I think it confuses things to not think of these as theme patterns - from the user perspective they are a theme patterns regardless of the source, but I will try and come up with a name for the constant that better reflects the situation

Reply:
> from the user perspective they are a theme pattern regardless of the source

Luckily, we're referring to a variable name here that isn't exposed to users, so we can afford to be accurate.

We could also argue that for most users they only see patterns, not user, theme, or core patterns etc.

Reply:
I don't think the new naming of `NON_THEME_PATTERN_SOURCES` introduced in https://github.com/WordPress/gutenberg/pull/55877/commits/316bc22c83c83f9f5c6ea8a2b381dd35e462f756 is any better than `PATTERN_CORE_SOURCES`. As discussed, the source is still the pattern directory not the theme, so `NON_THEME_PATTERN_SOURCES` should still include `pattern-directory/theme` in my view.

Reply:
Have renamed the constant to `NON_THEME_PATTERN_SOURCES`, which makes the nature of it a bit clearer in the contexts it is used I think, eg. 

```js
const selectThemePatterns = createSelector(
...
    .filter( ( pattern ) => ! NON_THEME_PATTERN_SOURCES.includes( pattern.source ))
...
)

```

Reply:
Dan's suggested name `EXCLUDED_LIBRARY_SOURCES` sidesteps most of the discussed issue by naming the array by its purpose. It might require an extra tweak to an [inline comment and variable](https://github.com/WordPress/gutenberg/pull/55877/files#diff-44a736aefd65eb0bc771fc2c753df374d275be921dede4d7da2e746565c98b9bR44-R45) but that's neither here nor there.

Reply:
have updated that inline comment and variable